flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java

```diff
@@ -60,26 +63,27 @@
 	private static final Logger LOG = LoggerFactory.getLogger(ContinuousFileMonitoringFunction.class);
 
 	/**
-	 * The minimum interval allowed between consecutive path scans. This is applicable if the
-	 * {@code watchType} is set to {@code PROCESS_CONTINUOUSLY}.
+	 * The minimum interval allowed between consecutive path scans.
+	 * <p><b>NOTE:</b> Only applicable to the {@code PROCESS_CONTINUOUSLY} mode.
 	 */
-	public static final long MIN_MONITORING_INTERVAL = 100l;
+	public static final long MIN_MONITORING_INTERVAL = 1l;
 
 	/** The path to monitor. */
 	private final String path;
 
-	/** The default parallelism for the job, as this is going to be the parallelism of the downstream readers. */
+	/** The parallelism of the downstream readers. */
 	private final int readerParallelism;
 
 	/** The {@link FileInputFormat} to be read. */
 	private FileInputFormat<OUT> format;
 
-	/** How often to monitor the state of the directory for new data. */
+	/** The interval between consecutive path scans. */
 	private final long interval;
 
 	/** Which new data to process (see {@link FileProcessingMode}. */
 	private final FileProcessingMode watchType;
 
+	/** The maximum file modification time seen so far. */
 	private Long globalModificationTime;
```

Comment:
Why is this `Long` by the way? You would usually avoid the unboxing by using `long`. The `volatile` flag is to ensure that the thread always sees the latest update to the `globalModificationTime`. If you don't use `volatile`, caching might prevent the thread from seeing the latest update.